src/main/java/org/acme/infinispanclient/HogwartsMagicSocket.java

```diff
@@ -0,0 +1,83 @@
+package org.acme.infinispanclient;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.websocket.OnClose;
+import javax.websocket.OnError;
+import javax.websocket.OnOpen;
+import javax.websocket.Session;
+import javax.websocket.server.ServerEndpoint;
+
+import org.acme.infinispanclient.model.HPMagic;
+import org.acme.infinispanclient.service.DataLoader;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.query.api.continuous.ContinuousQuery;
+import org.infinispan.query.api.continuous.ContinuousQueryListener;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.infinispan.client.runtime.Remote;
+
+@ServerEndpoint("/harry-potter/hogwarts")
+@ApplicationScoped
+public class HogwartsMagicSocket {
+
+   private static final Logger LOG = LoggerFactory.getLogger("HogwartsMagicSocket");
+
+   @Inject
+   @Remote(DataLoader.HP_MAGIC_NAME)
+   private RemoteCache<String, HPMagic> magic;
+
+   private Map<Session, ContinuousQueryListener<String, HPMagic>> listeners = new ConcurrentHashMap<>();
+
+   @OnOpen
+   public void onOpen(Session session) {
+      LOG.info("Session opened");
+      ContinuousQuery<String, HPMagic> continuousQuery = Search.getContinuousQuery(magic);
```

Comment:
Probably a little clearer to put this right before line 61

Reply:
This wasn't moved either.

Reply:
moved